### ISSUE TITLE
Cannot disable Keybase Update notifications

### ISSUE THREAD

**Issue by nicjansma** [keybase/keybase-issues#3128]
Multiple times a day, I get this notification:

![image](https://user-images.githubusercontent.com/1004649/33583152-e22c8a6e-d926-11e7-814c-b9cb690f43d9.png)

I do not see any way of disabling these update notifications in the UI.

(this, along with https://github.com/keybase/keybase-issues/issues/2380#issuecomment-274571700 result in a very frustrating experience on Windows)

**Comment by maxtaco** [comment]
Sorry for the bad experience.

@zanderz can you investigate?

**Comment by zanderz** [comment]
"Snooze" should quiet it down for a day, is that not working?
You could click "Update Automatically" to let the updates proceed without interaction usually, it is pretty painless. We are updating desktop clients about every 2-3 weeks now, and it is usually better to run the newer version. So if you do update, even without enabling automatic updating, it should remain quiet for weeks at a time.

**Comment by zanderz** [comment]
But if you have already updated and believe this prompt is in error, please do a `keybase log send` and we'll investigate.

**Comment by nicjansma** [comment]
@zanderz Snooze seems to quiet it down for a day.  So every morning when I hop on my computer, I see a reminder from Keybase that I haven't updated.  I don't want to update Keybase right now, and I don't want to be notified every day.  Instead of Keybase being a nice quiet app that runs in the background and is there when I need it, I associate it with frustrating popups that I have to deal with every day.

Since the app does not have an option to disable these notifications, I have removed it from startup.  Since the app does not have an option to disable running on startup, I have to manually remove the startup keys from the registry.

(Un)luckily for me, the next time I decide to update the app, it will helpfully re-add the registry keys for me (https://github.com/keybase/keybase-issues/issues/2380#issuecomment-274571700).

**Please** add options for controlling those behaviors.

**Comment by zanderz** [comment]
Options to allow users to refuse updates indefinitely, with no prompting? I don't think we will ever add that.

Simply updating would have been far easier than removing the startup shortcuts, and would have kept it quiet for weeks.

**Comment by nicjansma** [comment]
Not an option to refuse updates -- an option to disable _any_ update notifications.  I don't like apps that expect my attention with popups.  Instead, you could let me check for updates on my own time from a menu option -- when I actually open the app (not just because it's running in the tray).

These are user-friendliness things that will frustrate users and can cause them to uninstall your app if not addressed (e.g. https://github.com/keybase/keybase-issues/issues/2380#issuecomment-274577563)

**Comment by ronindesign** [comment]
I'm uninstalling Keybase for this reason.

I don't leave things that call home on "auto".
Recently, there have been several updates close together, so despite manually updating, I keep getting "new update" pop-ups, so it seems like this is a lose-lose, trying to force users into always auto-updating.

Maybe a re-examination of the target market is in order... this is something I would expect from M$ not a modern tech company.

**Comment by richdougherty** [comment]
On Windows 10 for me I get a nag everyday *even if I update*. I assume this is because the update fails for some reason. Updates failed for the Feb and April updates at least. I'll try again with the next version. Apologies that I'm not providing much info here; I'm too busy to investigate, but I did want to add my data point that updates aren't working in my case.

As a final UI point, you may want to consider making the update information easier to parse so we know if this is a new update (and so I know whether to retry an update or just snooze again until tomorrow). Two concrete ideas to make it easier to parse the updates: give the update a human readable title such as "April 2018 Update" (I see the the date is embedded in the version tag but it's not easy to scan); don't start every update with the same text: "Design: Some new colors and icons", instead mention the main headline change at the top.

**Comment by zanderz** [comment]
@richdougherty Can you do a `keybase log send` at the CMD window so we can see why you still get pestered? There were recently multiple updates in one week but it is rare.

**Comment by nicjansma** [comment]
I'm now getting update notifications even if I close the Keybase client in my Windows 10 tray.

It appear these two process stick around and _oh so helpfully_ notify me of Keybase updates every day:

![image](https://user-images.githubusercontent.com/1004649/39097543-bc58c42a-462b-11e8-9839-12f6cd6dedaa.png)

I have a lot of applications installed in Win 10, and none of them annoy me with update notifications as much as Keybase.

I *don't* care about keeping Keybase constantly up to date if I'm not using it daily.  When I need to use Keybase, I will manually check for updates.  Stop annoying your users with these constant notifications, or **give us an option to disable the notification**.

**Comment by zanderz** [comment]
@nicjansma There is an option under advanced settings to not start keybase at startup. If you use that and quit via the tray menu widget, there shouldn't be any running processes, otherwise there is a bug.

**Comment by nicjansma** [comment]
I got another popup update notification today, even with no `keybase.exe` processes running.  The process showing the notification was `prompter.exe`.

What is triggering this?  I don't see anything in Scheduled Tasks or via Autoruns.

**Comment by zanderz** [comment]
@nicjansma If you have set the advanced option to not start at startup and rebooted, and still got a popup, there is some bug - can you do a `keybase log send`?

Not all keybase processes are named keybase.exe - you can see them all at `%localappdata%\Keybase`

**Comment by gbaker-blackbirdit** [comment]
I'd settle for an update window that doesn't take focus and completely disrupt what you're working on.  If you're presenting, or playing a game, or watching a movie, Keybase will rudely alarm you about an update.  An OS native notification or non-modal window that doesn't take the focus would do just fine.

**Comment by meiskam** [comment]
6 months later this is still an issue, nag screens every day.  please add an option to not check for updates, like every other program has

**Comment by soderluk** [comment]
This is really annoying! I've snoozed the updates today three times choosing "7 Days" as option. It's more like 7 minutes!
Also, this interrupts EVERYTHING I'm doing, be it in-game or anything else. Most annoying is when I'm in-game, and suddenly the game goes away and up comes keybase updater window.
Thinking of uninstalling and stop using this altogether, and stop advertising this as a good product.

Platform: Windows 10
Keybase GUI Version: 2.5.1-20180808131430+bbc9797a57

**Comment by darrylyeo** [comment]
Still happening on the macOS version. Very disruptive, appears multiple times a day. I'll get to it later, you don't have to tell me thrice!

Reminds me of another notorious software update popup:

![A new version of Java is available!](https://i.imgur.com/v9sCspU.png)

Please, a simple desktop notification and/or a message within the app window would suffice. I know you have the capability to do so:

![In-app notifications](https://user-images.githubusercontent.com/13970789/46820131-4db05880-cd3a-11e8-8626-9a279f62532c.png)

**Comment by NarniaT** [comment]
Updates are coming like zombies, this is a really bad decision guys. Please STOP it. You can simply notify the user about the updates when the app is opened! This will be a really bad reputation for you if it lasts like this.

**Comment by damianb** [comment]
Always wonderful when the update nag also decides to *steal focus* from a fullscreen application.  

Stop it! You don't have any right to steal focus over an update nag.  Be polite, pop up a notification in the tray, but don't open a window that steals focus.  That's a cardinal sin, a UX failure, and counterproductive to your end user.

**Comment by ronindesign** [comment]
Y'all, they clearly don't care. This is a design choice from them and it shows their intentions clearly.

Be thankful you get a unique insight into how they think and their dev practices and move on now to another product instead of investing in Keybase. 

Much worse decisions will follow in the future, but don't be surprised, they'll be exactly in line with the same reasoning that allows for this design decision. Just get out before you've invested too much. :)

**Comment by meiskam** [comment]
Good idea, I've just moved my data away and uninstalled because of this issue

**Comment by Ullaakut** [comment]
Uninstalling as well, the answers on this thread are an obvious sign that they don't take this issue seriously and are forcing users into enabling auto updates on purpose. I'm going to subscribe to this thread though, in case that changes one day.

In the meantime, big 👎, because this could have been awesome.

**Comment by spuder** [comment]
Also getting these pop ups multiple times per day. Please STAHP!

**Comment by webworker01** [comment]
Found related information here, seems that after an update this auto startup watchdog is reinstalled https://github.com/keybase/keybase-issues/issues/2380#issuecomment-274571700

**Comment by nakkurt** [comment]
Sad to see that over 1.5 years later this still hasn't been addressed with a simple user-controlled setting to turn off unwanted notifications. 
Big 👎

**Comment by jmhobrien** [comment]
What a pain. It interrupts whatever I'm doing. I need to uninstall.

**Comment by zapman449** [comment]
For MacOS: System Preferences -> Notifications -> Find Keybase in the left, and set the "keybase alert style" to "None".  No more popups seen in early testing.  

Might have to play with the lower checkboxes for keybase as well.  I only have "Show in Notification Center" checked.  Everything else is unchecked.

**Comment by fresheneesz** [comment]
I've uninstalled keybase because of this. A cryptography and security application should be aware of security implications of autoupdates and forcing users to update. Security applications should only be updated when a user reviews the update and is convinced the update is safe - otherwise its far more likely for a malicious update would infect many more users. Nagging users to update is contrary to this philosophy. 

Its also a pain in the ass. I would rather not use keybase than be bugged by popups for it to update. Its shameful that your team refuses to fix this problem.

**Comment by ronindesign** [comment]
Came back years later to give Keybase another try, looks like they still refuse to accommodate sane UX practice. Guess I'll skip installing to see what's new and maybe circle back in a few more years...
